Apps/WebClient/src/ClientApp/src/router.ts

```diff
@@ -241,7 +284,15 @@ router.beforeEach(async (to, from, next) => {
         // Make sure that the route accepts the current state
         const currentUserState = calculateUserState();
         logger.debug(`current state: ${currentUserState}`);
-        if (to.meta.validStates.includes(currentUserState)) {
+        const isValidState = to.meta.validStates.includes(currentUserState);
+        const availableModules = getAvailableModules();
+        const hasRequiredModules =
+            to.meta.requiredModules === undefined
+                ? true
+                : to.meta.requiredModules.every((val: string) =>
+                      availableModules.includes(val)
+                  );
+        if (isValidState && hasRequiredModules) {
             next();
         } else {
```

Comment:
if goint to timeline is not the best then maybe
```
else if (isValidState)
{
...
}
else {
   next({ path: "/unauthorized" }); // or notfound
}

Reply:
updated to direct user to unauthorized page when Dependent is disabled